### ISSUE TITLE
[BUG] Invalid database credentials for `WPLoader` still results in exit code `0`

### ISSUE THREAD

**Issue by johnbillion** [lucatume/wp-browser#551]
**Environment**
OS: Mac & Linux  
PHP version: 7.4  
Installed Codeception version: 4.1.27  
Installed wp-browser version: 3.0.17  
WordPress version: 5.8  
Local development environment: Nginx  
WordPress structure and management: default  

**Can you perform the test manually?**
N/A

**Codeception configuration file**    

```yaml
paths:
  tests: tests
  output: tests/_output
  data: tests/_data
  support: tests/_support
  envs: tests/_envs
actor_suffix: Tester
extensions:
  enabled:
    - Codeception\Extension\RunFailed
params:
  - tests/.env
```

**Suite configuration file**

```yaml
actor: WpunitTester
modules:
  enabled:
    - WPLoader
  config:
    WPLoader:
      wpRootFolder: "tests/wordpress"
      configFile: "vendor/autoload.php"
      dbName: "%WP_TESTS_DB_NAME%"
      dbHost: "%WP_TESTS_DB_HOST%"
      dbUser: "%WP_TESTS_DB_USER%"
      dbPassword: "INCORRECT_PASSWORD_HERE"
```

**Describe the bug**

When the database credentials for `WPLoader` are invalid -- for example the password is incorrect -- running the test suite results in an exit code of `0`. This means your tests will pass in a CI environment even though no tests run.

**Output**

The database connection error message and PHP warnings from WordPress core are shown:

```
PHP Warning:  mysqli_real_connect(): (HY000/1045): Access denied for user 'root'@'localhost' (using password: NO) in tests/wordpress/wp-includes/wp-db.php on line 1653
PHP Warning:  mysqli_real_connect(): (HY000/1045): Access denied for user 'root'@'localhost' (using password: NO) in tests/wordpress/wp-includes/wp-db.php on line 1653

Warning: mysqli_real_connect(): (HY000/1045): Access denied for user 'root'@'localhost' (using password: NO) in tests/wordpress/wp-includes/wp-db.php on line 1653

wp_die called
Message : <p><code>Access denied for user 'root'@'localhost' (using password: NO)</code></p>
<h1>Error establishing a database connection</h1>
**snip**
```

**To Reproduce**

Create a vanilla integration test suite but set an incorrect password for the database connection, then run the suite.

**Expected behavior**

An exit code other than `0` is produced.

**Comment by lucatume** [comment]
Thanks @johnbillion for reporting the issue and providing the information.
I will look into this and push a fix, I will update the issue when done.

**Comment by lucatume** [comment]
I've pushed a possible fix I've not extensively tested on `dev-master`, I will come back to this and test it more, but any early feedback is welcome.

**Comment by lucatume** [comment]
Should be fixed in release `3.0.18`, re-open if that is not the case.
